sycl/source/detail/device_image_impl.hpp

```diff
@@ -619,7 +619,7 @@ class device_image_impl {
 
   std::shared_ptr<kernel_impl> tryGetSourceBasedKernel(
       std::string_view Name, const context &Context,
-      const std::shared_ptr<kernel_bundle_impl> &OwnerBundle,
+      std::shared_ptr<kernel_bundle_impl> &&OwnerBundle,
```

Comment:
Why not raw pointer/reference instead? Is it because you want to keep the PR manageable in size and that subgraph of the codebase is left to a subsequent change?

Reply:
What I meant is that you'll `shared_from_this` in a place where copy is created before this change.

Reply:
> we should definitely take into account that C++11 move semantics is a difficult subject.

Exactly! No `move`s is just easier for everyone ;)